config/nginx.conf.erb

```diff
@@ -2,6 +2,10 @@ daemon off;
 #Heroku dynos have at least 4 cores.
 worker_processes <%= ENV['NGINX_WORKERS'] || 4 %>;
 
+set_real_ip_from 127.0.0.1;
+real_ip_header X-Forwarded-For;
+real_ip_recursive on;
```

Comment:
As far as I can tell, Heroku doesn't really have good docs on any of this.  Their [public docs](https://devcenter.heroku.com/articles/http-routing#heroku-headers) just say "`X-Forwarded-For`: the originating IP address of the client connecting to the Heroku router".  I found [this](https://stackoverflow.com/questions/18264304/get-clients-real-ip-address-on-heroku/18517550#18517550) on Stack Overflow, which says: "The router doesn't overwrite `X-Forwarded-For`, but it does guarantee that the real origin will always be the last item in the list."

Based on that, I think `real_ip_recursive` should be off (the default).  The [nginx docs](http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from) say that with this off: "the original client address [...] is replaced by the last address sent in the request header".

Reply:
I figured I'd leave recursive on as we add more proxies in the future, but you're right that the behavior is the same with it on or off right now.